bootstore/src/schemes/v0/state.rs

```diff
@@ -0,0 +1,594 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! State for the V0 protocol state machine
+
+use super::fsm::StateHandler;
+use super::fsm_output::{ApiError, ApiOutput};
+use super::messages::{Envelope, Request, RequestType};
+use super::state_learned::LearnedState;
+use super::state_learning::LearningState;
+use super::state_uninitialized::UninitializedState;
+use super::{fsm_output::Output, state_initial_member::InitialMemberState};
+use crate::{
+    trust_quorum::{LearnedSharePkgV0, RackSecret, SharePkgV0},
+    Sha3_256Digest,
+};
+use derive_more::From;
+use serde::{Deserialize, Serialize};
+use sha3::{Digest, Sha3_256};
+use sled_hardware::Baseboard;
+use std::collections::{BTreeMap, BTreeSet};
+use std::fmt::Debug;
+use uuid::Uuid;
+use zeroize::{Zeroize, ZeroizeOnDrop};
+
+/// A number of clock ticks from some unknown epoch
+pub type Ticks = usize;
```

Comment:
Nit: maybe make this explicitly a `u64`, or add a static assertion to check that `size_of::<usize>() == size_of::<u64>()`?

Reply:
No longer relevant. The new code just uses `Instant` and `Duration`. I realized that slow tickers or missed ticks could cause inaccuracy so removed the use of ticks as time in [98ada27](https://github.com/oxidecomputer/omicron/pull/3385/commits/98ada27b53d4429463cde94cc91bedec0bc4ef98)